### DISCUSSION TITLE
Working Group - Quarkus SSG => Roq

### DISCUSSION THREAD

**Discussion by ia3andy** [quarkusio/quarkus#41309] [Design Discussions]
Hello folks,

I am starting a new initiative to allow Static Site Generation with Quarkus (similar to Hugo, Jekyll, Lume, ...).

Quarkus already provides most of the pieces to create great web applications (https://quarkus.io/guides/web).

I recently added https://github.com/quarkiverse/quarkus-roq. It will allow generating a static website out of any Quarkus application (it starts the app, fetch all the configured pages and assets, generate a static website and stop), it already works but it is still very alpha.

**What's missing?** we now need to incrementally add the toolkit to ease the process of creating static content through Quarkus:
- [x] Static Data 
- [x] Markdown/Asciidoc and frontmatter
- [x] SEO
- [x] Image processing (https://github.com/quarkiverse/quarkus-web-bundler/issues/42)
- [x] Easy to configure routing

**To go further:**
- Compat with tools like https://frontmatter.codes/
- Roq GitHub action
- Dev-UI integrated headless CMS (to edit md/asciidoc on the fs)

This will allow to develop the content using Quarkus dev-mode, and then generate (on CI) for Github Pages or similar when it's ready.

Bonus, everything added will benefit any "non-static" Quarkus app and any static Quarkus app could also go back to being non static.

This effort is now tracked using a "Focus Group" (temporary wording) project: https://github.com/orgs/quarkiverse/projects/6

This is a great opportunity to participate in a fun focus group and be involved with the Quarkus community, if anyone is interested in being a part of this, please reach out to me 🚀

There will be small, medium, bigger features to develop with any level of involvement. Participating could just be giving thoughts and discussing things..

**Comment by ia3andy** [comment]
List of contributors already interested in being part of this effort, comment with +me and I'll add you to the list:
@mcruzdev
@melloware 
@edewit 
@mkouba 
@ia3andy 
@maxandersen 

PS: I might be able to create some "Quarkus Roq" tee shirts for those involved 🤩

**Reply by maxandersen** [reply]
+me. 

The idea of type safe data without requiring classes intrigues me:)

**Comment by ia3andy** [comment]
@liquidnya I thought you might be interested?

**Reply by melloware** [reply]
@tmulle thought you might be interested.

**Comment by ia3andy** [comment]
I added a poll on Zulip to choose a new name: https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/statiq/near/446537942

**Reply by ia3andy** [reply]
Roq it is
  <img src="https://github.com/quarkiverse/quarkus-roq/assets/2223984/0eb6642e-8dd3-4def-abb7-f63062ae755b" width="400" >
